gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -909,48 +1005,133 @@ gst_tensor_filter_transform (GstBaseTransform * trans,
+    } else if (out_trans_data->allocate_in_invoke) {
       /* prepare memory block if successfully done */
-      out_mem[i] = mem = gst_tensor_filter_get_wrapped_mem (self,
-          out_tensors[i].data, out_tensors[i].size);
+      out_trans_data->mem[i] = mem = gst_tensor_filter_get_wrapped_mem (self,
+          out_trans_data->tensors[i].data, out_trans_data->tensors[i].size);
 
-      if (out_flexible) {
+      if (out_trans_data->is_flexible) {
         /* prepare new memory block with meta */
-        out_mem[i] = gst_tensor_meta_info_append_header (&out_meta[i], mem);
+        out_trans_data->mem[i] =
+            gst_tensor_meta_info_append_header (&out_trans_data->meta[i], mem);
         gst_memory_unref (mem);
       }
     }
 
     /* append the memory block to outbuf */
-    gst_tensor_buffer_append_memory (outbuf, out_mem[i],
+    gst_tensor_buffer_append_memory (outbuf, out_trans_data->mem[i],
         gst_tensors_info_get_nth_info (&prop->output_meta, i));
   }
+}
 
-  return GST_FLOW_OK;
+/**
+ * @brief non-ip transform. required vmethod of GstBaseTransform.
+ */
+static GstFlowReturn
+gst_tensor_filter_transform (GstBaseTransform * trans,
+    GstBuffer * inbuf, GstBuffer * outbuf)
+{
+  GstTensorFilter *self = GST_TENSOR_FILTER_CAST (trans);
+  GstTensorFilterPrivate *priv = &self->priv;
+  GstTensorFilterProperties *prop = &priv->prop;
+  guint i;
+  gint invoke_res = -1;
+  gboolean need_profiling;
+  GstFlowReturn retval = GST_FLOW_OK;
+  FilterTransformData *in_trans_data = NULL;
+  FilterTransformData *out_trans_data = NULL;
+  GstTensorMemory *invoke_tensors = NULL;
+
+  /* 0. Check all properties. */
+  retval = _gst_tensor_filter_transform_validate (trans, inbuf, outbuf);
+  if (retval != GST_FLOW_OK)
+    return retval;
+
+  /* 1. Get all input tensors from inbuf. */
+  in_trans_data = _gst_tensor_filter_transform_get_input_data (trans, inbuf);
+  if (!in_trans_data) {
+    return GST_FLOW_ERROR;
+  }
+
+  invoke_tensors =
+      _gst_tensor_filter_transform_get_invoke_tensors (trans, in_trans_data);
+  if (!invoke_tensors) {
+    goto mem_map_error;
+  }
+
+  /* 2. Prepare output tensors. */
+  out_trans_data = _gst_tensor_filter_transform_get_output_data (trans);
+  if (!out_trans_data) {
+    goto mem_map_error;
+  }
+
+  retval =
+      _gst_tensor_filter_transform_prepare_output_tensors (trans,
+      out_trans_data);
+  if (retval != GST_FLOW_OK) {
+    goto mem_map_error;
+  }
+
+  need_profiling = (priv->latency_mode > 0 || priv->throughput_mode > 0 ||
+      priv->latency_reporting);
+  if (need_profiling)
+    prepare_statistics (priv);
+
+  /* 3. Call the filter-subplugin callback, "invoke" */
+  GST_TF_FW_INVOKE_COMPAT (priv, invoke_res, invoke_tensors,
+      out_trans_data->tensors);
+  if (need_profiling) {
+    record_statistics (priv);
+    track_latency (self);
+  }
+
+  /* 4. Free map info and handle error case */
+  retval =
+      _gst_tensor_filter_transform_check_result (trans, in_trans_data,
```

Comment:
The function name is good.
How about changing it to ```_gst_tensor_filter_transform_check_invoke_result```? :smile:

Reply:
Updated.